data/src/main/java/com/example/data/UserPageRepositoryImpl.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.data
+
+
+import android.util.Log
+import com.example.data.remote.FollowerDataSource
+import com.example.data.remote.UserDataSource
+import com.example.domain.UserPageRepository
+import com.example.domain.model.Follower
+import com.example.domain.model.Item
+import com.example.domain.model.User
+import com.example.domain.model.UserPage
+import kotlinx.coroutines.flow.*
+import javax.inject.Inject
+
+class UserPageRepositoryImpl @Inject constructor(
+    private val followerDataSource: FollowerDataSource,
+    private val userDataSource: UserDataSource,
+) : UserPageRepository {
+
+    override fun getUserPageInfo(userName: String): Flow<UserPage> =
+        getSearchItems(userName)
+            .flatMapConcat { items ->
+                zip(items)
+            }
+
+    private fun zip(items: List<Item>): Flow<UserPage> = flow {
+        items.forEach { item ->
+            val user = getUser(item.login)
+            val followers = getFollowers(item.login)
+            val firstElement = user.firstOrNull()
+
+            if (firstElement != null) {
+                emit(UserPage(user.first(), followers.toList().flatten()))
+            }
```

Comment:
```suggestion
            firstElement?.let{
                emit(UserPage(user.first(), followers.toList().flatten()))
            }
```
스코프함수 어떠나요?? (코틀린 스럽게 쓰면 좋을 것 같아서입니당)